src/ansys/units/unit.py

```diff
@@ -8,18 +9,41 @@ class Unit:
     Parameters
     ----------
     Name: str,
-        Name of the unit or string chain of combined units.
+        Name of the unit or string chain of combined units
     Config: dict
-        dictionary of unit properties.
+        dictionary of unit properties
+    Dimensions: Dimensions
+        An instance of the Dimensions class.
+    Unit System: str
+        Define the unit system for base units of dimension,
+        default is SI.
 
     Returns
     -------
     Unit
-        Unit instance
+        Unit instance.
     """
 
-    def __init__(self, units: str, config: dict = None):
-        self._name = units
+    def __init__(
+        self,
+        units: str = None,
+        config: dict = None,
+        dimensions: ansunits.Dimensions = None,
+        unit_sys: str = None,
+    ):
+        if units and dimensions:
+            raise UnitError.EXCESSIVE_PARAMETERS()
+
+        if units:
+            self._name = units
+            dimensions = self._units_to_dim(units=units)
+            self._dimensions = ansunits.Dimensions(dimensions)
+        elif dimensions:
+            self._dimensions = dimensions
+            self._name = self._dim_to_units(dimensions=dimensions, unit_sys=unit_sys)
```

Comment:
I suppose that there is a requirement to pass dimensions AND unit_sys so you should write here:
`elif dimensions and unit_sys:`